packages/core/src/components/graphs/scatter.ts

```diff
@@ -251,6 +263,96 @@ export class Scatter extends Component {
 						)
 					);
 
+				// Find all the data points that are placed within the threshold along x and y axis.
+				const [x, y] = mouse(self.parent.node());
+				const displayData = self.model.getDisplayData();
+				const scaledData: {
+					domainValue: number;
+					rangeValue: number;
+					originalData: any;
+				}[] = displayData.map((d) => ({
+					domainValue: self.services.cartesianScales.getDomainValue(
+						d
+					),
+					rangeValue: self.services.cartesianScales.getRangeValue(d),
+					originalData: d,
+				}));
+
+				const dataPointsWithinThreshold: {
+					domainValue: number;
+					rangeValue: number;
+					originalData: any;
+				}[] = scaledData
+					.filter((d) =>
+						pointIsWithinThreshold(
+							d.domainValue,
+							d.rangeValue,
+							x,
+							y
+						)
+					)
+					.reduce((accum, currentValue) => {
+						if (accum.length === 0) {
+							accum.push(currentValue);
+							return accum;
+						}
+
+						// Store the first element of the accumulator array to compare it with current element being processed.
+						const xAccumValue = accum[0].domainValue;
+						const yAccumValue = accum[0].rangeValue;
+						const xDistanceToCurrentValue = Math.abs(
+							x - currentValue.domainValue
+						);
+						const yDistanceToCurrentValue = Math.abs(
+							y - currentValue.rangeValue
+						);
+						const xDistanceToAccumValue = Math.abs(x - xAccumValue);
+						const yDistanceToAccumValue = Math.abs(y - yAccumValue);
+
+						if (
+							xDistanceToCurrentValue > xDistanceToAccumValue ||
+							yDistanceToCurrentValue > yDistanceToAccumValue
+						) {
+							// If distance with current value is bigger than already existing value in the accumulator,
+							// skip current iteration.
+							return accum;
+						} else if (
+							xDistanceToCurrentValue < xDistanceToAccumValue ||
+							yDistanceToCurrentValue < yDistanceToAccumValue
+						) {
+							// CurrentValue data point is closer to mouse inside the threshold area, so reinstantiate array.
+							accum = [currentValue];
+						} else {
+							// CurrentValue is equal to already stored values,
+							// which means there's another match on the same coordinate.
+							accum.push(currentValue);
+						}
+
+						return accum;
+					}, []);
+
+				if (dataPointsWithinThreshold.length > 0) {
+					const rangeIdentifier = self.services.cartesianScales.getRangeIdentifier();
+					const tooltipData = dataPointsWithinThreshold
+						.map((d) => d.originalData)
+						.filter((d) => {
+							const value = d[rangeIdentifier];
+							return value !== null && value !== undefined;
+						});
+					// Show tooltip
+					self.services.events.dispatchEvent(Events.Tooltip.SHOW, {
+						hoveredElement,
+						multidata: tooltipData,
+						type: TooltipTypes.DATAPOINT,
+					});
+				} else {
+					// Show tooltip
+					self.services.events.dispatchEvent(Events.Tooltip.SHOW, {
+						hoveredElement,
+						type: TooltipTypes.DATAPOINT,
+					});
+				}
+
```

Comment:
a massive amount of logic for something so simple, there has to be an easier way to do this

Reply:
I used the same logic used in showRuler() method in carbon-charts/packages/core/src/components/axes/ruler.ts

Alternatively, I can select all circles with cx and cy within certain threshold of current hovered element to be included in multidata tooltip. That will be a lot simpler than this and effectively be doing the same thing.

Reply:
To further simplify, I can remove the reduce() logic and replace it with "threshold" set to the points.radius value. This way the multidata tooltip will shown for points that overlap.

Reply:
@theiliad I have updated the code to use points.radius as threshold to determine the points to include in multidata tooltip. So if the points are overlapping, they will be included in the tooltip. This allowed me to remove the complex logic in map.reduce call.

Reply:
no there's no need for any that.

we should be able to simply:
1. get domain value of highlighted point
2. get range value of highlighted point
3. find any matching datapoints in `displayData` which don't belong in the same datagroup
4. pass data into tooltip

Then as long as we pass `multiData` into the tooltip it should show those values 
![image](https://user-images.githubusercontent.com/14989804/83822020-f04ee500-a69d-11ea-9365-5c203b995f65.png)

Reply:
So we only want to show multidata tooltip if the domain and range values are exact match?

Reply:
@theiliad I have updated the code as per your recommendation. I don't consider datagroup while filtering because in my use case, the multiple points are in same datagroup but are different because of some other attributes.

Reply:
I do want mention that the downside of not using the threshold approach is that its possible that the domain or range values may be different by a small fraction (not visible in UI). So to the user it will still appear that there's 1 point and the UI won't show multidata tooltip on hover over the overlapping points.

Reply:
@theiliad Can you please review the latest changes?